src/rules/check-inject-decorator.rule.ts

```diff
@@ -0,0 +1,44 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+const createRule = ESLintUtils.RuleCreator(
+  (name) => `https://eslint.org/docs/latest/rules/${name}`
+);
+
+export type MessageIds =
+  | 'tokenDuplicatesType'
+  | 'typeIsInterface'
+  | 'propertyMissingInject';
+
+const defaultOptions: unknown[] = [];
+
+export default createRule({
+  name: 'check-inject-decorator',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Ensure proper use of the @Inject decorator',
+      recommended: 'recommended',
+    },
+    fixable: undefined,
```

Comment:
Making this rule fixable in some cases would be nice too, @tuxmachine  - but probably a work for another issue, since it's a bit harder to do it